asteroid/dsp/spatial.py

```diff
@@ -0,0 +1,65 @@
+import torch
+import torch.nn.functional as F
+
+
+def xcorr(ref, input, normalized=True, eps=1e-8):
+    """
+    Efficient cross-correlation computation between multi-channel signals.
+    Input tensors must be of shape: mic_channels, batch, seq_len.
```

Comment:
I wonder if it should expect batch, mic_channels, seq_len to keep having the same input format everywhere

Reply:
Yes, I think that would be much better!

Reply:
Agree. What do you think that right now it does not compute all possible pairs when both inputs are multi-channel ? We do not need that functionality right now but we might need it in future.

Reply:
I guess that's fine for now. Probably by using broadcasting it can be achieved quite cheap.